volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,437 @@
+                optional=False,
+            ),
+        ]
+
+    @staticmethod
+    def _get_pefile_obj(
+        context, pe_table_name: str, layer_name: str, base_address: int
+    ) -> Optional[pefile.PE]:
+        """
+        Attempts to pefile object from the bytes of the PE file
+
+        Args:
+            pe_table_name: name of the pe types table
+            layer_name: name of the process layer
+            base_address: base address of the module
+
+        Returns:
+            the constructed pefile object
+        """
+        pe_data = io.BytesIO()
+
+        try:
+            dos_header = context.object(
+                pe_table_name + constants.BANG + "_IMAGE_DOS_HEADER",
+                offset=base_address,
+                layer_name=layer_name,
+            )
+
+            for offset, data in dos_header.reconstruct():
+                pe_data.seek(offset)
+                pe_data.write(data)
+
+            pe_ret = pefile.PE(data=pe_data.getvalue(), fast_load=True)
+
+        except exceptions.InvalidAddressException:
+            pe_ret = None
+
+        return pe_ret
+
+    @staticmethod
+    def _find_through_exports(
+        context,
+        config_path: str,
+        mod_name: str,
+        vads: Tuple[int, int, int],
+        wanted_symbols: Dict[str, List[str]],
+        found_symbols: Dict[str, Tuple[str, int]],
+    ):
+        """
+        Attempts to resolve the symbols in `wanted_symbols` through export table analysis
+        """
+        pe_table_name = intermed.IntermediateSymbolTable.create(
+            context, config_path, "windows", "pe", class_types=pe.class_types
+        )
+
+        # for each process layer and VAD, construct a PE and examine the export table
+        for proc_layer_name, vad_start, _ in vads:
+            # we need a valid PE with an export table
+            pe_module = PESymbols._get_pefile_obj(
+                context, pe_table_name, proc_layer_name, vad_start
+            )
+            if not pe_module:
+                continue
+
+            pe_module.parse_data_directories(
+                directories=[pefile.DIRECTORY_ENTRY["IMAGE_DIRECTORY_ENTRY_EXPORT"]]
+            )
+            if not hasattr(pe_module, "DIRECTORY_ENTRY_EXPORT"):
+                continue
+
+            # walk the export table looking for symbols of interest
+            for export in pe_module.DIRECTORY_ENTRY_EXPORT.symbols:
+                # AttributeError throws on empty or ordinal-only exports
+                try:
+                    func = export.name.decode("ascii")
+                except AttributeError:
+                    continue
+
+                if func in wanted_symbols[mod_name]:
+                    address = export.address
+
+                    if mod_name not in found_symbols:
+                        found_symbols[mod_name] = []
+
+                    found_symbols[mod_name].append((func, vad_start + address))
+                    wanted_symbols[mod_name].remove(func)
```

Comment:
This is altering wanted_symbols that's a mutable type (so will affect future uses of the variable).  Is that right/intentional?

Reply:
Yes